src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -484,27 +487,27 @@ private static object CreateInstance([DynamicallyAccessedMembers(DynamicallyAcce
 
         [RequiresUnreferencedCode("Cannot statically analyze what the element type is of the object collection so its members may be trimmed.")]
         private static void BindCollection(
-            object collection,
+            object? collection,
             [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods)]
             Type collectionType,
             IConfiguration config, BinderOptions options)
         {
             // ICollection<T> is guaranteed to have exactly one parameter
             Type itemType = collectionType.GenericTypeArguments[0];
-            MethodInfo addMethod = collectionType.GetMethod("Add", DeclaredOnlyLookup);
+            MethodInfo? addMethod = collectionType.GetMethod("Add", DeclaredOnlyLookup);
 
             foreach (IConfigurationSection section in config.GetChildren())
             {
                 try
                 {
-                    object item = BindInstance(
+                    object? item = BindInstance(
                         type: itemType,
                         instance: null,
                         config: section,
                         options: options);
                     if (item != null)
                     {
-                        addMethod.Invoke(collection, new[] { item });
+                        addMethod?.Invoke(collection, new[] { item });
```

Comment:
If there is no `Add` method, this will now silently no longer do anything. Is that what we want to happen?

Reply:
We didn't seem to do anything before (there's an empty catch on the line bellow).